### DISCUSSION TITLE
PHP versions?

### DISCUSSION THREAD

**Discussion by miken32** [nathanheffley/laravel-slack-blocks#20] [General]
Forking this for use internally and was going to do some PRs back. Would you accept PRs that change the version requirements? Making PHP require ^8.0 would allow for a lot of code cleanup with constructor property promotion and various other conveniences.

**Comment by nathanheffley** [comment]
Definitely! I have been meaning to make sure there's an up to date version, but I don't have any projects using it currently so its hard to test.

**Comment by miken32** [comment]
Sounds good, I'll have a PR in the next week or two.

One other potentially breaking change I was looking to make is moving the helper classes to a different namespace. So `NathanHeffley\LaravelSlackBlocks\Blocks\Divider` instead of `NathanHeffley\LaravelSlackBlocks\Messages\SlackAttachmentBlockDivider`. I'd imagine they're generally accessed from the helper functions so this shouldn't be too much of an impact. We're planning to use most of the block types so will want helper classes/functions for them all. Having them separated and using some inheritance to account for repeated properties (i.e. `block_id` which we use commonly) will make things easier to work with.

**Reply by nathanheffley** [reply]
I don't see an issue with that. I agree people should be accessing them through the helper functions or else they probably know what they're doing and will be fine. A new major version for this is fine IMO. Like I said, I haven't been actively using this so if you have I am sure you have a good idea on what would be helpful 🙂

**Comment by miken32** [comment]
lol, lmao, etc.

https://github.com/laravel/slack-notification-channel/releases/tag/v3.0.0
https://laravel.com/docs/10.x/notifications#formatting-slack-notifications

**Reply by nathanheffley** [reply]
Oh wow they finally did it! 😄 This package was only created originally as a patch because the core team didn't want to support the new format even when it was well known that the old method was being deprecated, I'm glad they finally came around.

In light of this, I will not be accepting a PR to upgrade this package because it looks like it will take minimal effort to update from my package to the official one.

**Reply by miken32** [reply]
Yup, agreed. Going to migrate all the work I did for this PR over to a PR on their package to implement all the missing blocks/elements. Have a good one!
